timemachine/fe/single_topology.py

```diff
@@ -1241,77 +1184,7 @@ def assert_chiral_consistency_and_validity(
-    """Verify that a core and forcefield would allow for valid chiral endstates.
-
-    Refer to `check_chiral_validity` for definition of validity.
-
-    Raises
-    ------
-        ChiralConversionError
-            If chiral end states are incompatible for the given core and forcefield
-        DummyGroupAssignmentError
-            If unable to find a chirally-valid dummy group assignment with respect to a placeholder forcefield
-    """
-    find_chirally_valid_dummy_groups = make_find_chirally_valid_dummy_groups(mol_a, mol_b)
-    dummy_groups = find_chirally_valid_dummy_groups(core)
-    if dummy_groups is None:
-        raise DummyGroupAssignmentError("Unable to find chirally-valid dummy group assignment")
-    dummy_groups_ab, dummy_groups_ba = dummy_groups
-    check_chiral_validity = make_check_chiral_validity(mol_a, mol_b, forcefield)
-    check_chiral_validity(core, dummy_groups_ab, dummy_groups_ba)
-
-
-def make_check_chiral_validity(
-    mol_a: Chem.Mol, mol_b: Chem.Mol, forcefield: Forcefield
-) -> Callable[[NDArray, Dict[int, FrozenSet[int]], Dict[int, FrozenSet[int]]], None]:
-    setup_end_state_harmonic_bond_and_chiral_potentials_ab = make_setup_end_state_harmonic_bond_and_chiral_potentials(
-        mol_a, mol_b, forcefield, verify=False
-    )
-    setup_end_state_harmonic_bond_and_chiral_potentials_ba = make_setup_end_state_harmonic_bond_and_chiral_potentials(
-        mol_b, mol_a, forcefield, verify=False
-    )
-
-    def check_chiral_validity(
-        core: NDArray,
-        dummy_groups_ab: Dict[int, FrozenSet[int]],
-        dummy_groups_ba: Dict[int, FrozenSet[int]],
-    ):
-        """Verify that a core and dummy groups would allow for valid chiral endstates.
-
-        Refer to `check_chiral_validity` for definition of validity.
-
-        Raises
-        ------
-            ChiralConversionError
-                If chiral end states are incompatible for the given core and forcefield.
-        """
-        atom_map = AtomMapMixin(mol_a, mol_b, core)
-        bond_pot_src, chiral_atom_pot_src, _ = setup_end_state_harmonic_bond_and_chiral_potentials_ab(
-            core, atom_map.a_to_c, atom_map.b_to_c, dummy_groups_ab
-        )
-        bond_pot_dst, chiral_atom_pot_dst, _ = setup_end_state_harmonic_bond_and_chiral_potentials_ba(
-            core[:, ::-1], atom_map.b_to_c, atom_map.a_to_c, dummy_groups_ba
-        )
-
-        src_chiral_restr_idx_set = ChiralRestrIdxSet(chiral_atom_pot_src.potential.idxs)
-        dst_chiral_restr_idx_set = ChiralRestrIdxSet(chiral_atom_pot_dst.potential.idxs)
-
-        check_chiral_validity_impl(
-            atom_map,
-            src_chiral_restr_idx_set,
-            dst_chiral_restr_idx_set,
-            bond_pot_src.potential.idxs,
-            bond_pot_dst.potential.idxs,
-        )
-
-    return check_chiral_validity
+    _assert_chiral_consistency(src_chiral_restr_idx_set, dst_chiral_restr_idx_set, src_bond_idxs, dst_bond_idxs)
```

Comment:
nit: Is there a reason not to just merge this with the above?

Reply:
willfix